compat/src/suspense.js

```diff
@@ -0,0 +1,143 @@
+import { Component, createElement, unmount } from 'preact';
+import { removeNode } from '../../src/util';
+
+export function catchRender(error, component) {
+	// thrown Promises are meant to suspend...
+	if (error.then) {
+		for (; component; component = component._ancestorComponent) {
+			if (component._childDidSuspend) {
+				component._childDidSuspend(error);
+				return true;
+			}
+		}
+	}
+
+	return false;
+}
+
+function removeDom(vnode) {
+	let tmp = vnode._component;
+	if (tmp != null) {
+		if (tmp = tmp._prevVNode) removeDom(tmp);
+	}
+	else if ((tmp = vnode._children) && vnode._lastDomChild) {
+		for (let i = 0; i < tmp.length; i++) {
+			if (tmp[i]) removeDom(tmp[i]);
+		}
+	}
+
+	if (tmp = vnode._dom) {
+		removeNode(tmp);
+	}
+}
+
+// having custom inheritance instead of a class here saves a lot of bytes
+export function Suspense(props) {
+	// we do not call super here to golf some bytes...
+	this._suspensions = [];
+}
+
+// Things we do here to save some bytes but are not proper JS inheritance:
+// - call `new Component()` as the prototype
+// - do not set `Suspense.prototype.constructor` to `Suspense`
+Suspense.prototype = new Component();
+
+/**
+ * @param {Promise} promise The thrown promise
+ */
+Suspense.prototype._childDidSuspend = function(promise) {
+	// saves 5B
+	const _this = this;
+
+	_this._suspensions.push(promise);
+	let len = _this._suspensions.length;
+
+	const suspensionsCompleted = () => {
```

Comment:
Should these be part of the `prototype`? Thinking that these could introduce to many closures? Most of the refs are based on `scope` so, I believe these could do it?

Reply:
Well the issue is that the func needs to access `len` that is only available here and not on the proto. Any idea how to solve this? Function binding?